src/main/java/io/openliberty/tools/common/plugins/config/ServerConfigDocument.java

```diff
@@ -532,9 +522,15 @@ private static void parseDocumentsInDirectory(File directory, ArrayList<Document
      * @throws IOException
      * @throws SAXException
      */
-    private static Document parseDocument(File file) throws FileNotFoundException, IOException, SAXException {
-        InputStream is = new FileInputStream(file.getCanonicalPath());
-        return parseDocument(is);
+    private static Document parseDocument(File file) throws FileNotFoundException, IOException {
+        try (FileInputStream is = new FileInputStream(file)) {
```

Comment:
The `parseDocument(InputStream in)` method already closes the input stream. Is it going to cause a problem that both this method and that one will try to close the stream? Or does it handle that fine?

Reply:
Will have to do some research on this.
It seems there's a difference between manually calling `close()` where I think it no-ops if already closed, vs auto-closing might have a different behavior.

Reply:
The close() on `FileInputStream` does nothing if it's already closed. And `InputStream` close() will also effectively do nothing if called twice.

To be safe, I also refactored some code that called these methods to try to prevent resource leaks from unclosed streams.